File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(str) {
    switch (true) {
        case isSilent(str):
            return 'Fine. Be that way!';
        case isShouting(str):
            return 'Whoa, chill out!';
        case isQuestion(str):
            return 'Sure.';
        default:
            return 'Whatever.'    
    }
    
    function isShouting(str) {
        return (str == str.toUpperCase()) && (str.search(/[A-Za-z]/) != -1);
    }
    
    function isQuestion(str) {
        return str.endsWith('?');
    }
    
    function isSilent(str) {
        return str.trim().length == 0;
    }
};


module.exports = Bob;
// test = '';
// 
// console.log(test == '');`

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
        String.ends_with?(input, "?") -> "Sure."
        String.trim(input) == "" -> "Fine. Be that way!"
        all_upcase?(input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end
  
  @spec all_upcase?(String.t()) :: boolean
  def all_upcase?(input_string) do
    String.match?(input_string, ~r/^[^a-z]+$/) && String.match?(input_string, ~r/\p{L}+/)
    # because I don't know how to get the regex correct…
    # The second match is a better than /[A-Z]+/
  end
  
end




